contributors/design-proposals/scheduling/pod-priority-resourcequota.md

```diff
@@ -0,0 +1,169 @@
+
+Thus, we decide to reuse the existing `Scopes` of `ResourceQuotaSpec` to provide a richer semantics for quota to cooperate with priority classes. 
+
+## Overview 
+
+This design doc introduces how to define a group of priority class scopes for the quota to match with and explains how quota enforcement logic is changed to apply the quota to pods with the given priority classes.
+
+## Detailed Design 
+
+### Expected behavior of ResourceQuota
+
+Four new values of `scopes` will be defined for `ResourceQuota` to describe its relationship with priority class name:
+
+| Scope | Description |
+| ----- | ----------- |
+| PriorityClassNameIn:priorityName1:priorityName2 | Match `kind=Pod` where `spec.priorityClassName in (priorityName1, priorityName2)` |
+| PriorityClassNameNotIn:priorityName1:priorityName2 | Match `kind=Pod` where `spec.priorityClassName notin (priorityName1, priorityName2)` |
+| PriorityClassNameExists |  Match `kind=Pod` where `spec.priorityClassName` |
+| PriorityClassNameNotExists |  Match `kind=Pod` where `!spec.priorityClassName` |
+
+*Example: Match all pods independent of priority*
+
+`$ kubectl create quota example --hard=count/pods=10`
+
+*Example: Match all pods where priority is foo*
+
+`$ kubectl create quota example --hard=count/pods=10 --scopes=PriorityClassNameIn:foo`
+
+*Example: Match all pods where there is no priority*
+
+`$ kubectl create quota example --hard=count/pods=10 --scopes=PriorityClassNameNotExists`
+
+To restrict usage of a particular priority without explicit quota, we can do the following by a new field named `matchScopes` introduced to `ResourceQuota` configuration.:
+
+*Example: Any pod with priority cluster-services or other-important-stuff must require explicit quota*
+
+```yaml
+apiVersion: apiserver.k8s.io/v1alpha1
+kind: AdmissionConfiguration
+plugins:
+- name: "ResourceQuota"
+  configuration:
+  apiVersion: resourcequota.admission.k8s.io/v1alpha1
+  kind: Configuration
+  limitedResources:
+  - resource: pods
+    matchScopes:
+    - "PriorityClassNameIn:cluster-services:other-important-stuff"
+```
+
+*Example: Any pod with priority set must require explicit quota*
+
+```yaml
+apiVersion: apiserver.k8s.io/v1alpha1
+kind: AdmissionConfiguration
+plugins:
+- name: "ResourceQuota"
+  configuration:
+  apiVersion: resourcequota.admission.k8s.io/v1alpha1
+  kind: Configuration
+  limitedResources:
+  - resource: pods
+    matchScopes:
+    - "PriorityClassNameExists"
+```
+
+There are some details to note:
+
+1. The absence of a quota for a particular priority class name means the quota is unbounded (unless configured otherwise by operator in admission control configuration).
+2. If multiple `ResourceQuota` apply to a Pod, the pod must satisfy all of them.
+
```

Comment:
Note for me, from disscuss with RH folks: esisting clusters is
```
PriorityClassIn: "cluster-service" will be suggested to used in existing cluster.

and creating a resourceQuota object in a few privileged namespases, which are authorized to use cluster-service class

With this configuration on existing clusters, clusters will start using this feature

Any new pod which is created from non-authorized namespace with cluster-service priority class will fail to admit
Rest all pod, with any priority class except cluster-service or without priority class, will be admitted even if they don't find matching quota.

This will provide the benefits of this feature on existing clusters, without requiring huge configurations.
```

Which I agree.